ILA: Issue while installing Workspace ONE Access Content Pack

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.1

### Module Version

2.10.0.1011

### PowerShell Version

7.3.6

### PowerCLI Version

13.1.0.21624340

### PowerVCF Version

2.4.1.1000

### Guest Operating System

windows server 2019

### Environment Details

_No response_

### Description

Issue while running Invoke-ILAdeployment at step Install Workspace ONE Access Content Pack

### Error or Debug Output

```powershell
[04-18-2024_09:57:04] [INFO] Install Workspace ONE Access Content Pack
 [04-18-2024_09:57:05] [INFO]  Error at Script Line 15766
 [04-18-2024_09:57:05] [INFO]  Relevant Command: $response = Invoke-RestMethod -Method 'GET' -Uri $Uri -Headers $ghHeaders
 [04-18-2024_09:57:05] [INFO]  Error Message: Cannot validate argument on parameter 'Uri'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 [04-18-2024_09:57:05] [ERROR] Exception calling "FromBase64String" with "1" argument(s): "The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters."
 [04-18-2024_09:57:05] [ERROR] System.Management.Automation.CmdletInvocationException: {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}
 ---> Microsoft.PowerShell.Commands.HttpResponseException: Response status code does not indicate success: 401 (Unauthorized).
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
 [04-18-2024_09:57:05] [ERROR] {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}
 [04-18-2024_09:57:05] [ERROR] System.Management.Automation.CmdletInvocationException: {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}
 ---> Microsoft.PowerShell.Commands.HttpResponseException: Response status code does not indicate success: 401 (Unauthorized).
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
 [04-18-2024_09:57:05] [ERROR] {"message":"Bad credentials","documentation_url":"https://docs.github.com/rest"}
 [04-18-2024_09:57:05] [ERROR] Cannot validate argument on parameter 'Uri'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 [04-18-2024_09:57:05] [ERROR] Cannot validate argument on parameter 'Uri'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 [04-18-2024_09:57:05] [ERROR] Cannot validate argument on parameter 'Uri'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
```

### Expected Behavior

There should not any errors while running Invoke-ILAdeployment

### Actual Behavior

Error while installing Workspace ONE Access Content Pack

### Steps to Reproduce

Run Invoke-ILAdeployment and it will fail at step Install Workspace ONE Access Content Pac

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -79,8 +79,9 @@
 - Enhanced `Invoke-UndoIomDeployment` cmdlet to remove the load balancer configured in VMware Aria Suite Lifecycle for VMware Aria Operations.
 - Enhanced `Invoke-UndoPcaDeployment` cmdlet to remove the load balancer configured in VMware Aria Suite Lifecycle for VMware Aria Automation.
 - Enhanced `Invoke-InvDeployment` cmdlet to add data collector and LDAP configuration functions.
-- Converted `aria-operations-notifications-vcf.csv` to 'aria-operations-notifications-vcf.json'.
-- Converted `aria-operations-notifications-srm.csv' to 'aria-operations-notifications-srm.json'.
+- Enhanced `Install-vRLIContentPack` cmdlet to compress the and stream the content pack JSON payload to VMware Aria Operations for Logs.
+- Converted `aria-operations-notifications-vcf.csv` to `aria-operations-notifications-vcf.json`.
+- Converted `aria-operations-notifications-srm.csv` to `aria-operations-notifications-srm.json`.
 
 ## v2.10.0
 

--- a/PowerValidatedSolutions.psd1
+++ b/PowerValidatedSolutions.psd1
@@ -8,10 +8,10 @@
 @{
 
     # Script module or binary module file associated with this manifest.
-    RootModule = 'PowerValidatedSolutions.psm1'
+    RootModule        = 'PowerValidatedSolutions.psm1'
 
     # Version number of this module.
-    ModuleVersion     = '2.11.0.1048'
+    ModuleVersion     = '2.11.0.1049'
 
     # ID used to uniquely identify this module
     GUID              = 'a6dfed7b-65d2-4da2-bdcc-7f3d3df9b75d'

--- a/PowerValidatedSolutions.psm1
+++ b/PowerValidatedSolutions.psm1
@@ -16925,7 +16925,7 @@ Function Enable-vRLIContentPack {
                                     if (-Not ($allContentPacks -match "403 (rate limit exceeded)")) {
                                         if ($contentPackUri = ($allContentPacks | Where-Object { $_.name -eq $contentPackFile }).url) {
                                             $contentPackResponse = Invoke-RestMethod -Method 'GET' -Uri $contentPackUri -Headers $ghHeaders
-                                            $json = $contentPackResponse | ConvertTo-Json -Depth 100 -Compress
+                                            $json = $contentPackResponse | ConvertTo-Json -Depth 100
                                             if (Get-vRLIContentPack | Where-Object { $_.name -eq $contentPackName }) {
                                                 Write-Warning "Installing Content Pack ($contentPackName v$contentPackVersion) to VMware Aria Operations for Logs ($($vcfVrliDetails.fqdn)), already exists: SKIPPED"
                                             } else {
@@ -52288,14 +52288,14 @@ Function Install-vRLIContentPack {
         Installs a Content Pack to VMware Aria Operations for Logs.
 
         .DESCRIPTION
-        The Install-vRLIContentPack cmdlet installed a content pack to VMware Aria Operations for Logs.
+        The Install-vRLIContentPack cmdlet installs a content pack to VMware Aria Operations for Logs.
 
         .EXAMPLE
         Install-vRLIContentPack -json $json
         This example installs a content pack to VMware Aria Operations for Logs from a JSON payload.
 
         .EXAMPLE
-        Insall-vRLIContentPack -update -json $json
+        Install-vRLIContentPack -update -json $json
         This example updates a content pack in VMware Aria Operations for Logs from a JSON payload.
 
         .PARAMETER update
@@ -52310,13 +52310,31 @@ Function Install-vRLIContentPack {
         [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [Switch]$update
     )
 
+    $uri = "https://$vrliAppliance/api/v2/content/contentpack"
+
+    if ($update) {
+        $uri += "?overwrite=true"
+    } else {
+        $uri += "?overwrite=false"
+    }
+
+    $headers = @{}
+    $vrliHeaders.GetEnumerator() | ForEach-Object { $headers[$_.Key] = $_.Value }
+    $headers['accept'] = '*/*'
+    $headers['Content-Encoding'] = 'gzip' # Indicate that the content is GZIP compressed
+
+    # Compress the JSON payload.
+    $compressedStream = New-Object System.IO.MemoryStream
+    $gzipStream = New-Object System.IO.Compression.GzipStream $compressedStream, ([System.IO.Compression.CompressionMode]::Compress)
+    $sw = New-Object System.IO.StreamWriter $gzipStream
+    $sw.Write($json)
+    $sw.Close()
+    $gzipStream.Close()
+    $byteArray = $compressedStream.ToArray()
+    $compressedStream.Close()
+
     Try {
-        if ($PsBoundParameters.ContainsKey("update")) {
-            $uri = "https://$vrliAppliance/api/v1/content/contentpack?overwrite=true"
-        } else {
-            $uri = "https://$vrliappliance/api/v1/content/contentpack"
-        }
-        Invoke-RestMethod -Method 'POST' -Uri $uri -ContentType 'application/octet-stream' -Headers $vrliHeaders -Body $json
+        Invoke-RestMethod -Method 'POST' -Uri $uri -Headers $headers -Body $byteArray -ContentType 'application/json'
     } Catch {
         Debug-ExceptionWriter -object $_
     }

